### ISSUE TITLE
Unexpected `MissingBreakInSwitch` for case: one switch inside another switch

### ISSUE THREAD

**Issue by xupyprmv** [teamed/qulice#692]
Here is an example:

```
public static int next(final int xxx,
    final int yyy) throws IOException {
    final int result;
    switch (xxx) {
        case 0:
            switch (yyy) {
                case 0:
                    result = 0;
                    break;
                case 1:
                    result = 1;
                    break;
                default:
                    result = 0;
            }
            break;
        case 1:
            result = 0;
            break;
        default:
            result = 1;
    }
    return result;
}
```

PMD validator result: 

`A switch statement does not contain a break (MissingBreakInSwitch)`

This is wrong result.

**Comment by krzyk** [comment]
@xupyprmv Yes, it is a problem, but I think that bigger one is that it allows nested switch statements, it shouldn't, just like nested ifs/fors/trys, **nested switch should be forbidden**.
This way you problem won't appear because it won't be allowed.
Could you modify the description to match that?

**Comment by xupyprmv** [comment]
@krzyk ticket caption and description was updated.

**Comment by krzyk** [comment]
@xupyprmv thanks

**Comment by krzyk** [comment]
@davvd valid bug

**Comment by davvd** [comment]
> @davvd valid bug 

@krzyk tag bug added to this issue

**Comment by davvd** [comment]
@xupyprmv thank you for the ticket reported, I topped your account for 15 mins, transaction `AP-2K859259RA747714R`

**Comment by krzyk** [comment]
@davvd this is postponed

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk thanks, I added "postponed" label

**Comment by davvd** [comment]
> @davvd this is postponed

@krzyk I will try to find someone else

**Comment by yegor256** [comment]
Fixed in 528588d88.

Added a new custom Checkstyle module `com.qulice.checkstyle.NestedSwitchCheck` that forbids `switch` statements nested inside other `switch` statements. It mirrors the existing `NestedIfDepth`/`NestedForDepth`/`NestedTryDepth` modules already enabled in qulice.

The original snippet from this ticket now produces:

```
Nested switch statements are not allowed
```

on the inner `switch (yyy)` line.

Changes:
- `src/main/java/com/qulice/checkstyle/NestedSwitchCheck.java` — new check; walks the `LITERAL_SWITCH` ancestor chain and logs a violation when it finds an enclosing `switch`.
- `src/main/resources/com/qulice/checkstyle/checks.xml` — wires the check into the default ruleset.
- `src/test/resources/com/qulice/checkstyle/ChecksTest/NestedSwitchCheck/` — `Valid.java` (two sibling switches, no violation), `Invalid.java` (nested `switch`, one violation), plus the usual `config.xml`/`violations.txt`.
- `src/test/java/com/qulice/checkstyle/ChecksTest.java` — registers the new check in the parameterized test source.
